addons/account/models/chart_template.py

```diff
@@ -211,8 +211,10 @@ def _load(self, template_code, company, install_demo):
 
         # Manual sync because disable above (delay_account_group_sync)
         AccountGroup = self.env['account.group'].with_context(delay_account_group_sync=False)
-        AccountGroup._adapt_accounts_for_account_groups(self.env['account.account'].search([]))
-        AccountGroup.search([])._adapt_parent_account_group()
+        AccountGroup._adapt_accounts_for_account_groups(check_all=True)
+        AccountGroup._adapt_parent_account_group(check_all=True)
```

Comment:
I don't think there exists a use case where we want to adapt accounts from other companies (except for branches setup)

so a more elegant solution would be to pass the company as parameter to _adapt_accounts_for_account_group() and _adapt_parent_account_group() so that the subquery is efficient.

```suggestion
        AccountGroup._adapt_accounts_for_account_groups(company=company.id)
        AccountGroup._adapt_parent_account_group(company.id)
```

Reply:
I will check, I'd like to be even more precise by selecting the record that were updates. It looks like an issue during tests, the setupclass is also doing a tryloading when setuping companies and it is slow.... but your solution could actually help there. Will make some experiment on another branch and keep you in touch once I have result, thanks for the review!